core/src/proxy/relayer_stage.rs

```diff
@@ -147,89 +134,136 @@ impl RelayerStage {
+        relayer_config: RelayerConfig,
+        cluster_info: Arc<ClusterInfo>,
         heartbeat_tx: Sender<HeartbeatEvent>,
-        expected_heartbeat_interval: Duration,
-        oldest_allowed_heartbeat: Duration,
         packet_tx: Sender<PacketBatch>,
-        relayer_endpoint: Endpoint,
         verified_packet_tx: Sender<(Vec<PacketBatch>, Option<SigverifyTracerPacketStats>)>,
-        trust_packets: bool,
         exit: Arc<AtomicBool>,
     ) {
-        const WAIT_FOR_FIRST_AUTH: Duration = Duration::from_secs(5);
-
-        let mut wait_count: usize = 0;
-        let mut stream_error_count: usize = 0;
-        let mut connect_error_count: usize = 0;
-        while access_token.lock().unwrap().value.is_empty() {
-            if exit.load(Ordering::Relaxed) {
-                return;
-            }
-            wait_count += 1;
-            datapoint_info!(
-                "relayer_stage-wait_for_auth",
-                ("wait_count", wait_count, i64)
-            );
-            sleep(WAIT_FOR_FIRST_AUTH).await;
-        }
+        const MAX_BACKOFF_S: u64 = 10;
+        const CONNECTION_TIMEOUT: Duration = Duration::from_secs(CONNECTION_TIMEOUT_S);
+        let mut backoff_sec: u64 = 1;
+        let mut error_count: u64 = 0;
 
-        let mut backoff = BackoffStrategy::new();
         while !exit.load(Ordering::Relaxed) {
-            match relayer_endpoint.connect().await {
-                Ok(channel) => {
-                    match Self::start_consuming_relayer_packets(
-                        &mut backoff,
-                        RelayerClient::with_interceptor(
-                            channel,
-                            AuthInterceptor::new(access_token.clone()),
-                        ),
-                        &heartbeat_tx,
-                        expected_heartbeat_interval,
-                        oldest_allowed_heartbeat,
-                        &packet_tx,
-                        &verified_packet_tx,
-                        trust_packets,
-                        &exit,
-                    )
-                    .await
-                    {
-                        Ok(_) => {}
-                        Err(e) => {
-                            stream_error_count += 1;
-                            datapoint_warn!(
-                                "relayer_stage-stream_error",
-                                ("count", stream_error_count, i64),
-                                ("error", e.to_string(), String),
-                            );
-                        }
-                    }
+            match Self::connect_auth_and_stream(
+                &relayer_config,
+                &cluster_info,
+                &heartbeat_tx,
+                &packet_tx,
+                &verified_packet_tx,
+                &exit,
+                &CONNECTION_TIMEOUT,
+            )
+            .await
+            {
+                Ok(_) => {
+                    backoff_sec = 0;
```

Comment:
i don't think this will ever return ok? im okay removing backoff_sec + waiting some small amount of time between reconnects (1s, 2s, or 5s)